packages/rpc-parsed-types/src/config-accounts.ts

```diff
@@ -0,0 +1,24 @@
+import { Address } from '@solana/addresses';
+
+type StakeConfigAccount = Readonly<{
+    slashPenalty: number;
+    warmupCooldownRate: number;
+}>;
+
+type ValidatorInfoAccount = Readonly<{
+    configData: unknown;
+    keys: {
+        pubkey: Address;
+        signer: boolean;
+    }[];
+}>;
+
+export type ConfigProgramAccount =
+    | Readonly<{
+          info: StakeConfigAccount;
+          type: 'stakeConfig';
+      }>
+    | Readonly<{
+          info: ValidatorInfoAccount;
+          type: 'validatorInfo';
+      }>;
```

Comment:
Wdyt about this?

```ts
type RpcParsedType<T> = Readonly<{
    info: T,
    type: string
}>;

type StakeConfigAccount = Readonly<{
    slashPenalty: number;
    warmupCooldownRate: number;
}>;

// stake config account
{
    const account = {
        info: {
            slashPenalty: 12,
            warmupCooldownRate: 0.25,
        },
        type: 'stakeConfig' as const,
    };
    account satisfies RpcParsedType<StakeConfigAccount>;
}
```

[Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBASmDGAFAhgJwM4QCYBVwQB4cA+KAXlgmSwHsA7AGxAIG8AoKTqASzoDMaALig4ANBy6hIw9MFS8A5mwC+xANxs2U6AGVgyANYQAwvT7cFAQXjwaAVzrBylavSasJndA2ToAFogQdMgMoMJ0dgC2AEYQqBpcUADuaJF2YKY0DLRJdDDIwBDhUbHxKuqaAPSVULKG0Lb8FlDINvaObOyJjbItbQ5OFF2JXLwCwsMjI96+AUEhYVAAjABM4lNTKahpGTRZOXkFRVAADAB0KwCs6xvKN1PawgDkdUamTQpPLehQPcD3UGUCS4rVsA1qBW46HMEB+cCQaEwuHwBD09Xe5is-UcFWUQA)

Reply:
Ohhh I see you are fixing the string value as part of the type. Hmmm. Maybe we can do both somehow?

Reply:
This should work.

```ts
type RpcParsedType<TType extends string, TInfo> = Readonly<{
    info: TInfo,
    type: TType
}>;


export type ConfigProgramAccount =
    | RpcParsedType<"stakeConfig", StakeConfigAccount>
    | RpcParsedType<"validatorInfo", ValidatorInfoAccount>;
```

Reply:
Nice, that definitely cleans it up a bit!